roles/awscommon/tasks/main.yml

```diff
@@ -16,3 +16,57 @@
   changed_when: false
   tags:
     - always
+
+- name: Retrieve vpc level instance facts
+  ec2_remote_facts:
+    filters:
+      instance-state-name: running
+      "tag:Stack": "{{ vpc_name }}"
+  register: vpc_instance_facts
+  tags:
+    - always
+
+- name: Retrieve target level instance facts
+  ec2_remote_facts:
+    filters:
+      instance-state-name: running
+      "tag:Stack": "{{ target }}"
+  register: target_instance_facts
+  tags:
+    - always
+
+- name: Add jumphost to runtime group for later usage
+  add_host:
+    name: "jump.{{ vpc_name }}.{{ opg_data.domain }}"
+    groups: "vpc, jumphost"
+    hostname: "jump.{{ vpc_name }}.{{ opg_data.domain }}"
+    ansible_user: "{{ ssh_prov_user|default('jenkins-agent') }}"
+    ansible_ssh_common_args: "{{ ssh_conn_opts| default(omit) }}"
+  tags:
+    - always
+
+- name: Add vpc level instances to runtime group for later usage
+  add_host:
+    name: "{{ item.tags.Name }}.internal"
+    groups: "{{ item.tags.Stack  + '_' + item.tags.Role }} "
+    hostname: "{{ item.private_ip_address }}"
+    ansible_user: "{{ ssh_prov_user|default('jenkins-agent') }}"
+    ansible_ssh_common_args: "{{ ssh_conn_opts| default(omit) }}"
+    create_time: "{{ item.launch_time | default(omit) }}"
+  with_items: "{{ vpc_instance_facts.instances | default([]) }}"
+  no_log: true
+  tags:
+    - always
+
+- name: Add target level instances to runtime group for later usage
```

Comment:
this breaks on existing environments where the role tag is missing from jump and master